pkg/chunk/storage/factory.go

```diff
@@ -60,16 +68,22 @@ func Opts(cfg Config, schemaCfg chunk.SchemaConfig) ([]chunk.StorageOpt, error)
 		}, memcache))
 	}
 
+	if len(caches) > 0 {
+		tieredCache = cache.NewTiered(caches)
+	} else {
+		tieredCache, err = cache.New(cfg.indexCache)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	opts, err := newStorageOpts(cfg, schemaCfg)
 	if err != nil {
 		return nil, errors.Wrap(err, "error creating storage client")
 	}
 
-	if len(caches) > 0 {
-		tieredCache := cache.Instrument("tiered-index", cache.NewTiered(caches))
-		for i := range opts {
-			opts[i].Client = newCachingStorageClient(opts[i].Client, tieredCache, cfg.IndexCacheValidity)
-		}
+	for i := range opts {
+		opts[i].Client = newCachingStorageClient(opts[i].Client, tieredCache, cfg.indexCache.DefaultValidity)
```

Comment:
Note to self: `cfg.indexCache.DefaultValidity` should be set to `cfg.IndexCacheValidity` for backwards compat.